core/trino-main/src/main/java/io/trino/execution/scheduler/PipelinedQueryScheduler.java

```diff
@@ -44,6 +44,7 @@
 import io.trino.execution.TableExecuteContextManager;
```

Comment:
maybe a better commit:

```
Handle source tasks skewness in ScaledWriterScheduler

When there is a big data skewness, there could be a bottleneck 
due to the skewed workers even if most of the workers are not 
over-utilized. Check both, weighted output buffer 
over-utilization rate and average output buffer over-utilization
rate, in case when there are many over-utilized small tasks
due to fewer not-over-utilized big skewed tasks.
```

Reply:
sure.